lib/src/Pos/Launcher/Configuration.hs

```diff
@@ -104,17 +109,35 @@ instance Default ConfigurationOptions where
 -- configuration at a given key.
 withConfigurations
     :: (WithLogger m, MonadThrow m, MonadIO m)
-    => ConfigurationOptions
+    => Maybe BlacklistPath
+    -> ConfigurationOptions
     -> (HasConfigurations => NtpConfiguration -> m r)
     -> m r
-withConfigurations co@ConfigurationOptions{..} act = do
-    logInfo ("using configurations: " <> show co)
-    Configuration{..} <- parseYamlConfig cfoFilePath cfoKey
-    let configurationDir = takeDirectory cfoFilePath
-    withCoreConfigurations ccCore configurationDir cfoSystemStart cfoSeed $
-        withUpdateConfiguration ccUpdate $
-        withSscConfiguration ccSsc $
-        withDlgConfiguration ccDlg $
-        withTxpConfiguration ccTxp $
-        withBlockConfiguration ccBlock $
-        withNodeConfiguration ccNode $ act ccNtp
+withConfigurations mBlacklistPath cfo act = do
+    logInfo ("using configurations: " <> show cfo)
+    cfg <- parseYamlConfig (cfoFilePath cfo) (cfoKey cfo)
+    -- Here is where we could read and add a blacklist.
+    blacklist <- case mBlacklistPath of
+        Nothing -> pure mempty
+        Just fp -> liftIO $ readBlacklistSrcAddrs fp
+    let configDir = takeDirectory $ cfoFilePath cfo
+    withCoreConfigurations (ccCore cfg) configDir (cfoSystemStart cfo) (cfoSeed cfo) $
+        withUpdateConfiguration (ccUpdate cfg) $
+        withSscConfiguration (ccSsc cfg) $
+        withDlgConfiguration (ccDlg cfg) $
+        withTxpConfiguration (addBlacklist blacklist $ ccTxp cfg) $
+        withBlockConfiguration (ccBlock cfg) $
+        withNodeConfiguration (ccNode cfg) $ act (ccNtp cfg)
+
+addBlacklist :: Set Address -> TxpConfiguration -> TxpConfiguration
+addBlacklist bset tcfg =
+  tcfg { tcBlacklistSrcAddrs = Set.union (tcBlacklistSrcAddrs tcfg) bset }
+
+newtype BlacklistPath = BlacklistPath FilePath
+
+readBlacklistSrcAddrs :: BlacklistPath -> IO (Set Address)
+readBlacklistSrcAddrs (BlacklistPath fp) = do
+  res <- filter (not . null) . fmap Text.strip . lines <$> readFile fp
+  case partitionEithers $ map decodeTextAddress res of
+    (x:_, _) -> error $ "Error reading blacklist file: " <> x
+    ([], xs) -> pure $ Set.fromList xs
```

Comment:
this is a good place to use `AccValidation` imo, which would make this:

```haskell
case traverse (toAccValidation . over _Left (:[]) . decodeTextAddress) res of
    AccFailure errs -> error $ "Error reading blacklist file: \n" <> Text.unlines errs
    AccSuccess xs -> pure $ Set.fromList xs
```

Reply:
Yeah, `AccValidation` is nice, but I'd normally be hesitant to pull in a new library for a single usage in some code that is going to be removed once multi-signature addresses are working.

Reply:
Yeah. In any case we should IMO be printing out all the errors